dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -109,6 +109,18 @@ def authenticate
       return log_unauthorized(exception)
     end
 
+    # Schoology has multiple contexts that will launch LTI tools in an iframe.
+    # In this case, we will redirect to the iframe route, to prompt user to open
+    # in a new tab. This flow appends a 'new_tab=true' query param, so it will
+    # pass this block once the iframe "jail break" has happened.
+    if decoded_jwt[:iss] == Policies::Lti::LMS_PLATFORMS[:schoology][:issuer] && !params[:new_tab]
```

Comment:
Would it be worth making a policy method like `forces_iframe_launch?` similar to what [Juan did for `issuer_accepts_resource_link`](https://github.com/code-dot-org/code-dot-org/pull/57068/files#diff-4a6088c854aa5f6c3deca26139c544ed84030aebfe3ca8d679def7fde62ef0f3R135)? Would cover Schoology ever having another issue, like how Canvas has a separate test issuer, or future issuers that also force iframes. Might be YAGNI, idk.

Reply:
Yeah, I considered moving this to a policy method. I'll do it.